satellite/console/service.go

```diff
@@ -856,6 +897,100 @@ func (s *Service) GetBucketUsageRollups(ctx context.Context, projectID uuid.UUID
 	return s.store.UsageRollups().GetBucketUsageRollups(ctx, projectID, since, before)
 }
 
+// CreateMonthlyProjectInvoices creates invoices for all created projects on monthly basis.
+// Edge Dates are derived from the date parameter taking UTC year and month, then adding first
+// and last date of the month accordingly
+func (s *Service) CreateMonthlyProjectInvoices(ctx context.Context, date time.Time) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	// TODO: do we need auth here?
+	//auth, err := GetAuth(ctx)
+	//if err != nil {
+	//	return err
+	//}
+
+	utc := date.UTC()
+	startDate := time.Date(utc.Year(), utc.Month(), 1, 0, 0, 0, 0, time.UTC)
+	endDate := time.Date(utc.Year(), utc.Month()+1, 1, 0, 0, 0, -1, time.UTC)
+
+	// disallow invoice generation for future periods
+	if endDate.After(time.Now()) {
+		return errs.New("can't create invoices for future periods")
+	}
+
+	projects, err := s.store.Projects().GetAll(ctx)
```

Comment:
Would it not be possible to just query for projects within the timeframe instead of filtering it out afterwards?

Reply:
good point